docs/general-usage/model-types.rst

```diff
@@ -9,11 +9,39 @@ We created a number of built-in types for general use, but you can create your o
 using `Graphene <https://github.com/graphql-python/graphene/>`_ (Grapple's underlying library)
 and take advantage of Grapple's generic ``GraphQLField`` type.
 
+GraphQLField
+-------------
+.. module:: grapple.models
+.. class:: GraphQLField(field_name: str, field_type: type = None, required=None, **kwargs)
+
+    .. attribute:: field_name (str)
+
+        This is the name of the class property used in your model definition.
+
+    .. attribute:: field_type (type)
+
+        A Grapple model type such as ``GraphQLString`` or ``GraphQLBoolean``.
+
+    .. attribute:: required (bool=None)
+
+        Represents the field as non-nullable in the schema. This promises the client that it will have a value returned.
+
+    .. attribute:: kwargs
+
+        Useful keyword arguments:
+
+        * ``source`` (string)
+            You can pass a source string that is an attribute or method on the
+            class itself. If pointing to a method, all values from the instance
+            will be available within the `values` kwarg (`example <https://github.com/GrappleGQL/wagtail-grapple/blob/main/example/home/blocks.py#L99>`_).
```

Comment:
nitpick: if linking to code in the repo, use the commit hash as target may move.

having said that, this doc already includes code examples, so I suggest adding the snippet here

Reply:
further nitpick:

> If pointing to a method, all values from the instance will be available within the `values` kwarg

but only in a `StreamField` context. For `Page` or other models you only get access to the instance.